cli/src/cmd/forge/script/mod.rs

```diff
@@ -410,6 +416,72 @@ impl ScriptArgs {
         };
         Ok((func.clone(), data))
     }
+
+    /// Checks if the transaction is a deployment with a size above the `CONTRACT_MAX_SIZE`.
+    ///
+    /// If `self.broadcast` is enabled, it asks confirmation of the user. Otherwise, it just warns
+    /// the user.
+    fn check_contract_sizes(
+        &self,
+        transactions: Option<&VecDeque<TypedTransaction>>,
+        known_contracts: &BTreeMap<ArtifactId, ContractBytecodeSome>,
+    ) -> eyre::Result<()> {
+        if let Some(txes) = transactions {
+            for tx in txes {
+                if let Some(data) = tx.data() {
```

Comment:
To reduce rightward drift you can do something like:

```suggestion
        transactions.iter().flat_map(|txes| txes).filter(|tx| tx.data())
```

Reply:
can also add if data.len() > CONTRACT_MAX_SIZE and to == NameOrfAddress::Name in the filter?

Reply:
The `CONTRACT_MAX_SIZE` yeah, but not the `to` comparison. 

Inside `for`  it needs to figure out which is which (`CREATE` vs `CREATE2`) because of the 32 byte offset